kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt

```diff
@@ -36,7 +42,26 @@ class KSAnnotationImpl private constructor(private val annotationApplication: Kt
     }
 
     override val arguments: List<KSValueArgument> by lazy {
-        annotationApplication.arguments.map { KSValueArgumentImpl.getCached(it) }
+        val presentArgs = annotationApplication.arguments.map { KSValueArgumentImpl.getCached(it) }
+        val presentNames = presentArgs.mapNotNull { it.name?.asString() }
+        val absentArgs = analyze {
+            annotationApplication.classId?.toKtClassSymbol()?.let { symbol ->
+                symbol.getMemberScope().getConstructors().singleOrNull()?.let { constructor ->
+                    constructor.valueParameters.filter { valueParameter ->
+                        valueParameter.name.asString() !in presentNames
+                    }.mapNotNull { valueParameterSymbol ->
+                        valueParameterSymbol.getDefaultValue()?.let { constantValue ->
+                            KSValueArgumentImpl.getCached(
+                                KtNamedAnnotationValue(
+                                    valueParameterSymbol.name, KtConstantAnnotationValue(constantValue)
+                                )
+                            )
+                        }
+                    }
+                }
+            } ?: emptyList<KSValueArgument>()
+        }
+        presentArgs + absentArgs
```

Comment:
What if the args appear out of order or alternatively? E.g.,
```
annotation class A(
    val a1: Int = 1,
    val a2: Int = 2,
    val a3: Int = 3,
    val a4: Int = 4
)

@A(a3 = 1, a1 = 3)
```

Reply:
argument is a name value pair therefore ordering is not a factor, actually this is same with FE 1.0 implementation.